### PR TITLE
add properties and test for cewrapper

### PR DESCRIPTION
**Pull request by partouf**
TODO

- [x] Test + fix binary mode
- [x] Try to use ClangCompiler as a base for mingw-clang
- [x] Intel mode on MinGW Clang is disabled
- [ ] Libraries?
- [x] As soon as you use any headers, (iostream for example), execution fails
- [x] Copy paste to C

### CODE REVIEW

File: etc/cewrapper/compilers-and-tools.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+    "allowed_paths": [
+        {
+            "path": "Z:\\",
+            "rw": false
+        }
+    ],
+    "allowed_registry": []
+}
```
Comment (mattgodbolt) [inline etc/cewrapper/compilers-and-tools.json:9]: Some missing newlines at ends of files here (only noticed at github makes a meal out of them!)

**Comment by partouf** [comment]
MSVC CL.EXE writes to the temp folder, the lower privilege SAFER_LEVELID_CONSTRAINED does not work - even though the user I'm using has Full control over the temp folder (‪`C:\Users\<myuser>\AppData\Local\Temp`)

It's a little confusing, but the documentation (https://learn.microsoft.com/en-us/windows/win32/api/winsafer/nf-winsafer-safercreatelevel) does mention "regardless of the user rights of the user"

The AppLocker functions https://learn.microsoft.com/en-us/windows/security/threat-protection/windows-defender-application-control/applocker/applocker-functions

File: lib/parsers/asm-parser.ts [hunk 2]
```diff
@@ -615,7 +619,7 @@ export class AsmParser extends AsmRegex implements IAsmParser {
         const labelDefinitions: Record<string, number> = {};
         const dontMaskFilenames = filters.dontMaskFilenames;
 
-        let asmLines = asmResult.split('\n');
```
Comment (partouf) [inline lib/parsers/asm-parser.ts:618]: took me a while to notice this was the main issue for things not to show up in binary mode on windows :(

**Comment by partouf** [comment]
This clang does not support libc++, not entirely sure what to do with that information, as undoubtedly we'll get comments on that...

**Pull request merged**